### ISSUE TITLE
JS error when OverflowMenuItem click causes disabled state change

### ISSUE THREAD

**Issue by brunnerh** [IBM/carbon-components-svelte#714]
E.g. if i have an item that closes files, but is only enabled if files are opened:

![image](https://user-images.githubusercontent.com/834235/123959116-184a6e00-d9ae-11eb-95af-d2070e31d415.png)

```svelte
<ToolbarMenu icon={Menu20} iconDescription="Menu">
	<!-- ... -->
	<ToolbarMenuItem on:click={() => data = []}
		disabled={data.length == 0}>
		Close files
	</ToolbarMenuItem>
</ToolbarMenu>```
```

An error happens in this callback, because `ref` is `null`:

```js
// in /OverflowMenu/OverflowMenuItem.svelte
afterUpdate(() => {
  if (primaryFocus) {
    ref.focus();
  }
});
```

My suspicion is that the `disabled` change causes this callback to be executed after the menu has already been closed and the DOM discarded. The same could probably happen for the other properties as well.

Maybe just make that call defensive?

```js
afterUpdate(() => {
  if (primaryFocus && ref != null) {
    ref.focus();
  }
});
```

**Comment by metonym** [comment]
Good catch. Thanks for reporting this!

**Comment by metonym** [comment]
Fixed in v0.38.2
